src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_validate.c

```diff
@@ -82,6 +82,13 @@ int wm_agent_upgrade_validate_system(const char *platform, const char *os_major,
             }
         }
 
+        // Show an error message when upgrade a macOS (arm64) agent using an intel64 WPK package
+        if (arch) {
```

Comment:
We shouldn't do this. There is no necessity since there are no official WPK packages for MacOS ARM yet, and for `upgrade_custom` it should fail in the agent (there is no way to identify what the package contains). I think you need to identify why, for these systems, this `if` is not correct to allow the execution of the upgrade:
```c
if (!strcmp(platform, "windows") || (os_major && arch && (strcmp(platform, "ubuntu") || os_minor))) {
```

Reply:
We're going to filter the architecture provisionally. We'll remove this when we support WPK for macOS@ARM.